x/coin_test.go

```diff
@@ -324,3 +324,109 @@ func TestCoinSubtract(t *testing.T) {
 		})
 	}
 }
+
+func TestCoinDivide(t *testing.T) {
+	cases := map[string]struct {
+		total    Coin
+		pieces   int64
+		wantOne  Coin
+		wantRest Coin
+	}{
+		"split into one piece": {
+			total:    NewCoin(7, 11, "BTC"),
+			pieces:   1,
+			wantOne:  NewCoin(7, 11, "BTC"),
+			wantRest: NewCoin(0, 0, "BTC"),
+		},
+		"split into two pieces with no rest": {
+			total:    NewCoin(4, 0, "BTC"),
+			pieces:   2,
+			wantOne:  NewCoin(2, 0, "BTC"),
+			wantRest: NewCoin(0, 0, "BTC"),
+		},
+		"split into two pieces with fractional division and no rest": {
+			total:    NewCoin(5, 0, "BTC"),
+			pieces:   2,
+			wantOne:  NewCoin(2, FracUnit/2, "BTC"),
+			wantRest: NewCoin(0, 0, "BTC"),
+		},
+		"split into two pieces with a leftover": {
+			total:    NewCoin(0, 3, "BTC"),
+			pieces:   2,
+			wantOne:  NewCoin(0, 1, "BTC"),
+			wantRest: NewCoin(0, 1, "BTC"),
+		},
+		"split into two pieces with a fractional division and a leftover": {
+			total:    NewCoin(1, 0, "BTC"),
+			pieces:   3,
+			wantOne:  NewCoin(0, FracUnit/3, "BTC"),
```

Comment:
pp: please always put the numbers here and don't do math for results as a general rule of thumb. You may otherwise accidentally replicate a rounding issue.